docs/howto/features/exam.md

```diff
@@ -0,0 +1,38 @@
+# Set up a hub for an exam
+
+We provide support for using a JupyterHub in a *controlled* physical
+space for exams. This is an extra paid feature where we charge hourly
+for responsiveness.
+
+This page documents what we do to prep, based on our prior experiences.
+
+1. Make sure the exact dates and times of the exam are checked well in
+   advance, and we have enough engineering coverage during this time period.
+   Engineers should also *test* their access to the infrastructure and the
+   hub beforehand, to make sure they can fix issues if needed.
+
+2. For the duration of the exam, all user pods must have a
+   [guaranteed quality of service class](https://kubernetes.io/docs/tasks/configure-pod-container/quality-service-pod/).
+   In practice, this means we have memory & cpu requests set to be the same
+   as guarantees. This is to ensure equity - no user should get more or less
+   resources than any other. It also improves reliability.
+
+   This usually increases cost too, so should be done no more than 12h before
+   the start of the exam. It should be reverted back soon after the exam
+   is done.
+
+3. The instructor running the exam should test out their exam on the hub,
+   and make sure that it will complete within the amount of resources assigned
+   to it. They should also make sure that the environment (packages, python
+   versions, etc) are set up appropriately. From the time they test this until
+   the exam is over, new environment changes are put on hold.
+
+4. We should pre-warm the cluster the hub is on before the start of the exam,
```

Comment:
I translate the practical steps for this to calculating how many nodes are needed for the user count we guarantee can startup fluently, and then reconfiguring a `min_nodes` configuration of some kind temporarily. This can be done from terraform or in other ways, and having guidance on that could also be relevant.

For this, we should probably also have the image configured in singleuser.image and prePuller.continuous.enabled to ensure all started nodes have the image that will be used ahead of users arriving to them.